packages/@eris-boiler/common/src/modules/loadable/map.ts

```diff
@@ -0,0 +1,163 @@
+    ...loadables: Array<T | Array<T> | string | Array<string>>
+  ): this {
+    this.toLoad = this.toLoad.concat(
+      loadables
+        .reduce<Array<T | string>>((ax, dx) => ax.concat(
+          Array.isArray(dx) ? dx : [ dx ],
+        ), []),
+    )
+
+    return this
+  }
+
+  public async load (): Promise<this> {
+    const loadables = await this.resolveToLoad()
+
+    for (const loadable of loadables) {
+      const key = loadable[this.key]
+      if (typeof key !== 'string') {
+        throw new LoadableBadKey(
+          loadable.filePath ?? '',
+          this.key.toString(),
+        )
+      }
+
+      this.set(key, loadable)
+      if (this.onLoad) {
+        this.onLoad(loadable)
+      }
+    }
+
+    return this
+  }
+
+  public async reload (): Promise<this> {
+    for (const { filePath } of this.values()) {
+      if (filePath) {
+        fileLoader.unload(filePath)
+      }
+    }
+    const oldToLoad = this.toLoad
+
+    this.toLoad = this.reloadPaths
+
+    const loadables = await this.resolveToLoad()
+
+    for (const loadable of loadables) {
+      const key = loadable[this.key]
+      if (typeof key !== 'string') {
+        throw new LoadableBadKey(
+          loadable.filePath ?? '',
+          this.key.toString(),
+        )
+      }
+
+      this.set(key, loadable)
+
+      const oldLoadableObject = this
+        .find((old) => old.filePath === loadable.filePath)
+
+      if (oldLoadableObject) {
+        this.delete(oldLoadableObject[this.key] as unknown as string)
+        if (this.onReload) {
+          this.onReload(oldLoadableObject, loadable)
+        }
+      } else if (this.onLoad) {
+        this.onLoad(loadable)
+      }
+    }
+
+    this.toLoad = oldToLoad
+
+    return this
+  }
+
+  /**
+   * Load data files.
+   * @param   path The path to the loadable file/directory.
+   * @returns      The loadable objects loaded from file.
+   */
+  protected async loadFiles (path: string): Promise<Array<T> | T> {
+    this.reloadPaths.push(path)
+    const file = await fs.stat(path)
+    if (file.isDirectory()) {
+      const loadables = await fileLoader.loadDirectory(path)
+      const res = []
+      for (const loadable of loadables) {
+        if (!(loadable instanceof this._constructor)) {
+          throw new LoadableTypeError(loadable.filePath)
+        }
+        res.push(loadable)
+      }
+
+      return res
+    } else {
+      const loadable = await fileLoader.load(path)
+      if (!loadable) {
+        throw new LoadableNotFoundError(path)
+      }
+      if (!(loadable instanceof this._constructor)) {
+        throw new LoadableTypeError(path)
+      }
+
+      return loadable
+    }
+  }
+
+  /**
+   * Resolve loadable, be it path or array.
+   * @param   loadable Parse a loadable to clean up any arrays or paths.
+   * @returns          The cleaned loadable(s).
+   */
+  private async resolveToLoad (): Promise<Array<T>> {
+    const res = await Promise.all(this.toLoad.map(async (loadable) =>
```

Comment:
nitpick but the map function doesn't have to be async

Reply:
I think that there will be a problem in the case if I don't call `loadFiles` ?

Because then the array passed to promise.all would not be all promises.